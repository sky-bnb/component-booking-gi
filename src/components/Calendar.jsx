import React from 'react';
import styled from 'styled-components';
import Day from './Day.jsx';

export const StyledCalendar = styled.table`
  padding: 0px 10px;
  border-collapse: collapse;
  display: flex;
  justify-content: center;
`;

const StyledRow = styled.tr`
  height: 41px;
  width: 288px;
`;

const buildMonth = (firstDayOfMonth, daysInMonth) => {
  const daysArray = Array(firstDayOfMonth);
  const monthArray = [];
  const dayPadding = { date: null, isBlank: true };

  daysArray.fill(dayPadding);

  for (let i = 1; i <= daysInMonth; i += 1) {
    daysArray.push({ date: i, isBlank: false });
  }

  const daysToPad = 7 - (daysArray.length % 7);

  for (let i = 0; i < daysToPad; i += 1) {
    daysArray.push(dayPadding);
  }

  for (let i = 0; i < daysArray.length / 7; i += 1) {
    const week = daysArray.slice(i * 7, (i + 1) * 7);
    monthArray.push(week);
  }

  return monthArray;
};

const Calendar = (props) => {
  const {
    calendarMonth, firstDayOfMonth, daysInMonth, available, isCheckinOpen, isCheckoutOpen,
    updateCheckinDate, updateCheckoutDate, checkinValue, checkoutValue,
  } = props;

  return (
    <StyledCalendar>
      <tbody>{ buildMonth(firstDayOfMonth, daysInMonth).map((week, index) => 
        <StyledRow key={index}>
          { week.map(({ date, isBlank }, index) => {
            return ( <Day available={available}
              calendarMonth={calendarMonth}
              date={date} key={index}
              isBlank={isBlank}
              isCheckinOpen={isCheckinOpen}
              isCheckoutOpen={isCheckoutOpen}
              checkinValue={checkinValue}
              checkoutValue={checkoutValue}
              updateCheckinDate={updateCheckinDate}
              updateCheckoutDate={updateCheckoutDate}
            />);
          })}
        </StyledRow>) }</tbody>
    </StyledCalendar>
  );
};

export default Calendar;
