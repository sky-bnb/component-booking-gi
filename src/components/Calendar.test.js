import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Calendar, { StyledCalendar } from './Calendar.jsx';

describe('Calendar Component', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Calendar />));

  it('renders the Calendar element', () => {
    expect(wrapper.find(StyledCalendar).length).toBe(1);
  });

  it('matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
