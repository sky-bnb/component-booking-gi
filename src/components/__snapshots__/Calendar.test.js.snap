// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Component matches the snapshot 1`] = `
<Calendar__StyledCalendar>
  <tbody>
    <Calendar__StyledRow
      key="0"
    >
      <Day
        date={null}
        isBlank={true}
        key="0"
      />
      <Day
        date={null}
        isBlank={true}
        key="1"
      />
      <Day
        date={null}
        isBlank={true}
        key="2"
      />
      <Day
        date={null}
        isBlank={true}
        key="3"
      />
      <Day
        date={null}
        isBlank={true}
        key="4"
      />
      <Day
        date={null}
        isBlank={true}
        key="5"
      />
      <Day
        date={null}
        isBlank={true}
        key="6"
      />
    </Calendar__StyledRow>
  </tbody>
</Calendar__StyledCalendar>
`;
